src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -698,85 +645,175 @@ void importRegularField( std::vector< vtkIdType > const & cellIds,
-        }
-        else
-        {
-          importRegularField( cellIds, vtkArray, wrapper );
-        }
+        // Skip - the user may have not enabled a particular physics model/solver on this dstRegion.
+        GEOSX_LOG_LEVEL_RANK_0( 1, GEOSX_FMT( "Skipping import of {} -> {} on {}/{} (field not found)",
+                                              vtkArray->GetName(), wrapperName, region.getName(), subRegion.getName() ) );
+        continue;
+      }
+      WrapperBase & wrapper = subRegion.getWrapperBase( wrapperName );
+
+      GEOSX_LOG_LEVEL_RANK_0( 1, GEOSX_FMT( "Importing field {} -> {} on {}/{}",
+                                            vtkArray->GetName(), wrapperName, region.getName(), subRegion.getName() ) );
+
+      if( materialWrapperNames.count( wrapperName ) > 0 && wrapper.numArrayDims() > 1 )
+      {
+        importMaterialField( cellIds, vtkArray, wrapper );
+      }
+      else
+      {
+        importRegularField( cellIds, vtkArray, wrapper );
       }
     }
   } );
 }
 
 void VTKMeshGenerator::importFields( DomainPartition & domain ) const
 {
+  GEOSX_LOG_RANK_0( GEOSX_FMT( "{} '{}': importing field data from mesh dataset", catalogName(), getName() ) );
+  GEOSX_ASSERT_MSG( m_vtkMesh, "Must call generateMesh() before importFields()" );
+
   // TODO Having CellElementSubRegion and ConstitutiveBase... here in a pure geometric module is problematic.
   ElementRegionManager & elemManager = domain.getMeshBody( this->getName() ).getMeshLevel( 0 ).getElemManager();
 
-  auto importFieldsForElementType = [&]( VTKCellType vtkCellType,
-                                         std::map< int, std::vector< vtkIdType > > const & cellBlocks )
+  std::vector< vtkDataArray * > const srcArrays = findArraysForImport( *m_vtkMesh, m_fieldsToImport );
+
+  for( auto const & typeRegions : m_cellMap )
   {
-    for( auto & cellBlock: cellBlocks )
+    for( auto const & regionCells : typeRegions.second )
     {
-      importFieldOnCellElementSubRegion( cellBlock.first, vtkCellType, cellBlock.second, elemManager, m_importableArrays );
+      importFieldOnCellElementSubRegion( regionCells.first,
+                                         typeRegions.first,
+                                         regionCells.second,
+                                         elemManager,
+                                         m_fieldNamesInGEOSX,
+                                         srcArrays );
     }
-  };
-
-  importFieldsForElementType( VTK_HEXAHEDRON, m_regionsHex );
-  importFieldsForElementType( VTK_TETRA, m_regionsTetra );
-  importFieldsForElementType( VTK_WEDGE, m_regionsWedges );
-  importFieldsForElementType( VTK_PYRAMID, m_regionsPyramids );
+  }
 
-  std::map< string, string_array > fieldNames;
-  for( vtkDataArray * vtkArray: m_importableArrays )
+  // this is needed to avoid breaking the unit test testVTKImport, in which CommunicationTools is not setup
+  if( MpiWrapper::commSize( MPI_COMM_GEOSX ) > 1 )
```

Comment:
I think I copied this accidentally from `PAMELAMeshGenerator`, where the unit test kind invokes field import. The if statement doesn't change anything otherwise, since `synchronizeFields()` is effectively a no-op for single-rank jobs.
The solution would be to initialize `CommunicationTools` in both tests, which requires constructing a `GeosxState`.

Reply:
> I will see if I can do it in this PR.

Step by step!